### PR TITLE
In xcode_backend.sh, build flutter assemble arguments as an array of quoted strings

### PR DESCRIPTION
**Pull request by zanderso**
Prior to this PR, some arguments such as `--sksl-bundle-path` are passed to `flutter assemble` without quotes such that a path with a space in it would be interpreted as two separate command line arguments, leading to various errors.

This PR assembles the argument array as a bash array of quoted strings, which makes it easier to conditionally add some arguments while ensuring that they are properly quoted.

This will help the benchmarks added in https://github.com/flutter/flutter/pull/86460 get closer to running.

### CODE REVIEW

**Review by jmagman — COMMENTED** [review]
Does this fail on master and pass on this PR if you change this:
https://github.com/flutter/flutter/blob/72ae6e236ea192a686b1c015f2a6b242682888cd/packages/flutter_tools/test/integration.shard/ios_content_validation_test.dart#L69

to:
```dart
 '--split-debug-info=foo with space/',
```

Adding a `--dart-define=` with spaces would also be a way to test this (and expand coverage).

macOS has the same issue:
https://github.com/flutter/flutter/blob/master/packages/flutter_tools/bin/macos_assemble.sh
tested in:
https://github.com/flutter/flutter/blob/72ae6e236ea192a686b1c015f2a6b242682888cd/packages/flutter_tools/test/integration.shard/macos_content_validation_test.dart#L47

**Review by jmagman — COMMENTED** [review]
Huh I'm surprised nothing complained about checking in those app_icon_*.png binaries.

**Comment by zanderso** [comment]
> Huh I'm surprised nothing complained about checking in those app_icon_*.png binaries.

They're there in the `ios/` subdir as well.

File: packages/flutter_tools/test/integration.shard/analyze_size_test.dart [hunk 1]
```diff
@@ -79,10 +83,58 @@ void main() {
...
+    final String devToolsCommand = result.stdout.toString()
+        .split('\n')
+        .firstWhere((String line) => line.contains(runDevToolsMessage));
+    final String commandArguments = devToolsCommand.split(runDevToolsMessage).last.trim();
+    final String relativeAppSizePath = outputFilePath.split('.flutter-devtools/').last.trim();
+
+    expect(commandArguments.contains('--appSizeBase=$relativeAppSizePath'), isTrue);
+    expect(codeSizeDir.existsSync(), true);
     expect(result.exitCode, 0);
-  }, skip: true); // Extremely flaky due to https://github.com/flutter/flutter/issues/68144
```
Comment (jmagman) [inline packages/flutter_tools/test/integration.shard/analyze_size_test.dart:85]: 👍

**Review by jmagman — APPROVED** [review]
> > Huh I'm surprised nothing complained about checking in those app_icon_*.png binaries.
> 
> They're there in the `ios/` subdir as well.

I noticed that, I was surprised by that too!

The build test on the new macos project passed:
```
▌16:45:03▐ RUNNING: cd examples/hello_world; ../../bin/flutter build macos --debug -v
```
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8841508241899792080/+/u/run_test.dart_for_build_tests_shard_and_subshard_3_4/stdout

**Pull request merged**